community/kernel/src/test/java/org/neo4j/kernel/impl/api/KernelTransactionsTest.java

```diff
@@ -255,11 +256,26 @@ public void shouldBeAbleToSnapshotDuringHeavyLoad() throws Throwable
                 {
                     try ( KernelTransaction transaction = getKernelTransaction( transactions ) )
                     {
-                        parkNanos( MILLISECONDS.toNanos( random.nextInt( 3 ) ) );
-                        if ( snapshots.get( threadIndex ) == null )
+                        KernelTransactionsSnapshot snapshot = null;
+                        try
                         {
-                            snapshots.set( threadIndex, transactions.get() );
                             parkNanos( MILLISECONDS.toNanos( random.nextInt( 3 ) ) );
+                            if ( snapshots.get( threadIndex ) == null )
+                            {
+                                snapshot = requireNonNull( transactions.get(), "impossible" );
```

Comment:
I think what we are afraid of is `transactions` being `null`, not the result for `.get()` being `null`. Although that might be a nice thing to check as well!
Maybe do something like:
```
requireNonNull( transactions );
snapshot = requireNonNull( transactions.get() );
```

Reply:
Done